update/process/shardBlock.go

```diff
@@ -0,0 +1,173 @@
+package process
+
+import (
+	"math/big"
+
+	"github.com/ElrondNetwork/elrond-go/core"
+	"github.com/ElrondNetwork/elrond-go/core/check"
+	"github.com/ElrondNetwork/elrond-go/data"
+	"github.com/ElrondNetwork/elrond-go/data/block"
+	"github.com/ElrondNetwork/elrond-go/hashing"
+	"github.com/ElrondNetwork/elrond-go/marshal"
+	"github.com/ElrondNetwork/elrond-go/process"
+	"github.com/ElrondNetwork/elrond-go/sharding"
+	"github.com/ElrondNetwork/elrond-go/update"
+)
+
+// ArgsNewShardBlockCreatorAfterHardFork defines the arguments structure to create a new shard block creator
+type ArgsNewShardBlockCreatorAfterHardFork struct {
+	ShardCoordinator   sharding.Coordinator
+	TxCoordinator      process.TransactionCoordinator
+	PendingTxProcessor update.PendingTransactionProcessor
+	ImportHandler      update.ImportHandler
+	Marshalizer        marshal.Marshalizer
+	Hasher             hashing.Hasher
+}
+
+type shardBlockCreator struct {
+	shardCoordinator   sharding.Coordinator
+	txCoordinator      process.TransactionCoordinator
+	pendingTxProcessor update.PendingTransactionProcessor
+	importHandler      update.ImportHandler
+	marshalizer        marshal.Marshalizer
+	hasher             hashing.Hasher
+}
+
+// NewShardBlockCreatorAfterHardFork creates a shard block processor for the first block after hardfork
+func NewShardBlockCreatorAfterHardFork(args ArgsNewShardBlockCreatorAfterHardFork) (*shardBlockCreator, error) {
+	if check.IfNil(args.ImportHandler) {
+		return nil, update.ErrNilImportHandler
+	}
+	if check.IfNil(args.Marshalizer) {
+		return nil, update.ErrNilMarshalizer
+	}
+	if check.IfNil(args.Hasher) {
+		return nil, update.ErrNilHasher
+	}
+	if check.IfNil(args.ShardCoordinator) {
+		return nil, update.ErrNilShardCoordinator
+	}
+	if check.IfNil(args.TxCoordinator) {
+		return nil, update.ErrNilTxCoordinator
+	}
+	if check.IfNil(args.PendingTxProcessor) {
+		return nil, update.ErrNilPendingTxProcessor
+	}
+	if check.IfNil(args.ImportHandler) {
```

Comment:
This check is already done in line 38. Remove it from here

Reply:
done